amida-tech/greyscale
# Task/inba 348 uoa soft delete

#### What's this PR do?
Soft deletes subjects from greyscale. 

#### Related JIRA tickets:
https://jira.amida-tech.com/browse/INBA-348

#### How should this be manually tested?
1. Run Greyscale, Auth-service, Indaba-client and survey-service. 
2. Create a project, add subject. 
3. Delete a subject
4. Check the DB to ensure that it was deleted from all appropriate tables: `ProductUOA`, `UnitOfAnalysis`, `Tasks`

Note: You should only be able to delete subjects that do not have completed tasks. 
#### Any background context you want to provide?
You might need to re-seed your DB by running one of the integration tests:
`node_modules/.bin/mocha test/user.integration.js --bail -t 10000`

If testing with POSTMAN, the request body would need to be passed i as a raw json for DELETE
#### Screenshots (if appropriate):
